### PR TITLE
Add cbt plugin

### PR DESCRIPTION
**Pull request by jodersky**
Pending https://github.com/cvogt/cbt/pull/539

### CODE REVIEW

File: build_tools.md [hunk 1]
```diff
@@ -34,10 +34,10 @@ The following chart describes how well ENSIME supports each build tool:
...
+| docs fetching          | ✔                   | ✔          |                           |                             | ✔                     | ✔                            |                  |
 | launch manager         | ✔                   |            |                           |                             |                       |                              |                  |
 | debug launcher         | ✔                   |            |                           |                             |                       |                              |                  |
 | single file compile    | ✔                   |            |                           |                             |                       |                              |                  |
-| formatting             | ✔                   |            |                           |                             |                       | ✔                            |                  |
+| formatting             | ✔                   | ✔          |                           |                             |                       | ✔                            |                  |
```
Comment (jodersky) [inline build_tools.md]: Not sure what "formatting" here means. CBT has a scalafmt plugin, does that qualify?

File: build_tools/cbt.md [hunk 2]
```diff
@@ -4,4 +4,41 @@ order: 3
...
+```
+cbt ensime
+```
+
+This configuration will contain the current cbt `BaseBuild` and all of
+its dependent `BaseBuild`s as projects.
+
+## Configure
+The ENSIME configuration generated by CBT is fully customizable. Check
+out the [plugin's main method](todo) for an overview of the various
```
Comment (jodersky) [inline build_tools/cbt.md]: once the upstream PR is merged, I'll add a link here

File: build_tools.md (hunk 1 above)
Comment (fommil) [inline build_tools.md]: Yup!

**Comment by jodersky** [comment]
It's been merged upstream :tada: ! Please consider this pr as ready, now

**Comment by fommil** [comment]
w00t!

**Pull request merged**